QuizApp/QuizResultViewController.swift

```diff
@@ -14,19 +14,25 @@ class QuizResultViewController: UIViewController {
     
     private var container: UIView!
     private var gradientLayer: CAGradientLayer!
+    private var quizId: Int!
     private var correct: Int!
     private var total: Int!
+    private var time: Double!
     private var resultLabel: UILabel!
     private var finishButton: UIButton!
     
+    private var ns: NetworkService!
+    
     private let colorBackgroundLight = UIColor(red: 0.45, green: 0.31, blue: 0.64, alpha: 1.00)
     private let colorBackgroundDark = UIColor(red: 0.15, green: 0.18, blue: 0.46, alpha: 1.00)
     private let colorButtonText = UIColor(red: 0.39, green: 0.16, blue: 0.87, alpha: 1.00)
     private let buttonFont = UIFont(name: "SourceSansPro-Bold", size: 16.0)
     
-    init(correct: Int, total: Int) {
+    init(quizId: Int, correct: Int, total: Int, time: Double) {
```

Comment:
You could create a struct to encapsulate all this data and then just pass in an instance of the struct. Example:
```
struct QuizResult {
    let quizId: Int
    let correct: Int
    let total: Int
    let time: Double
}
```

Your init would be much shorter

```
init(result: QuizResult) {
    self.result = result
    ...
}
```